src/pyop/provider.py

```diff
@@ -405,7 +408,8 @@ def _verify_client_authentication(self, request_body, http_headers=None):
         if http_headers is None:
             http_headers = {}
         token_request = dict(parse_qsl(request_body))
-        verify_client_authentication(self.clients, token_request, http_headers.get('Authorization'))
+        client_id = verify_client_authentication(self.clients, token_request, http_headers.get('Authorization'))
```

Comment:
Could be simplified to `token_request['client_id'] = verify_client_authentication(self.clients, token_request, http_headers.get('Authorization'))`.
(Removing the next line.)